### PR TITLE
SV-759 your organisation is not approved to assess this version

### PR DESCRIPTION
**Pull request by ben1stone**

### CODE REVIEW

**Review by CodeWolf86 — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.AssessorService.Web/Controllers/CertificateVersionController.cs [hunk 1]
```diff
@@ -88,12 +96,10 @@ private async Task<IActionResult> LoadViewModel(string view)
         [HttpPost(Name = "Version")]
         public async Task<IActionResult> Version(CertificateVersionViewModel vm)
         {
-            return await SaveViewModel(vm,
-                returnToIfModelNotValid: "~/Views/Certificate/Version.cshtml",
-                nextAction: RedirectToAction("Declare", "CertificateDeclaration"), action: CertificateActions.Version);
+            return await SaveViewModel(vm, nextAction: RedirectToAction("Declare", "CertificateDeclaration"), action: CertificateActions.Version);
```
[the remark this answers is not in the document]
Comment (ben1stone) [reply src/SFA.DAS.AssessorService.Web/Controllers/CertificateVersionController.cs:99]: I was thinking that before this change there were two paths that returned the view set in the parameter whereas now there's only one as were are redirecting away if the selection is valid but not approved. I can change that back though

File: src/SFA.DAS.AssessorService.Web/Controllers/CertificateVersionNotApprovedController.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        [HttpGet("not-approved")]
+        public async Task<IActionResult> NotApprovedToAssess(bool? redirectToCheck = false)
+        {
+            var sessionString = SessionService.Get(nameof(CertificateSession));
+            if (sessionString == null)
+            {
+                return RedirectToAction("Index", "Search");
+            }
+
+            var standardUId = SessionService.Get("AttemptedStandardVersion");
```
[the remark this answers is not in the document]
Comment (ben1stone) [reply src/SFA.DAS.AssessorService.Web/Controllers/CertificateVersionNotApprovedController.cs]: Done

File: src/SFA.DAS.AssessorService.Web/Controllers/CertificateVersionNotApprovedController.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+using SFA.DAS.AssessorService.Application.Api.Client.Clients;
+using SFA.DAS.AssessorService.Web.Infrastructure;
+using SFA.DAS.AssessorService.Web.ViewModels.Certificate;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.AssessorService.Web.Controllers
+{
+    [Authorize]
+    [Route("certificate/version")]
+    public class CertificateVersionNotApprovedController : CertificateBaseController
```
[the remark this answers is not in the document]
Comment (ben1stone) [reply src/SFA.DAS.AssessorService.Web/Controllers/CertificateVersionNotApprovedController.cs]: yeah your right, it was only using the session but I have injected that separately

File: src/SFA.DAS.AssessorService.Web/Views/Certificate/VersionNotApproved.cshtml [hunk 4]
```diff
@@ -0,0 +1,42 @@
+﻿@model SFA.DAS.AssessorService.Web.ViewModels.Certificate.CertificateVersionNotApprovedViewModel;
+
+@{
+    ViewBag.Title = $"Error: Your Organisation is not currently approved to assess version {Model.AttemptedVersion}";
+    Layout = "_DashboardLayout";
+}
+
+<a class="govuk-back-link" asp-controller="CertificateVersion" asp-action="Version" asp-route-redirectToCheck="@(Model.BackToCheckPage)">Back</a>
```
[the remark this answers is not in the document]
Comment (ben1stone) [reply src/SFA.DAS.AssessorService.Web/Views/Certificate/VersionNotApproved.cshtml]: I've updated this. I've created a routeData variable that will add the route value only if it's true. So it won't appear when it isn't required

**Pull request merged**